projects/ngx-tour-md-menu/src/lib/tour-anchor.directive.ts

```diff
@@ -64,39 +64,97 @@ export class TourAnchorMatMenuDirective
     // Ignore step.placement
     if (!step.preventScrolling) {
       if (!withinviewport(this.element.nativeElement, { sides: 'bottom' })) {
-        (<HTMLElement>this.element.nativeElement).scrollIntoView(false);
+        (this.element.nativeElement as HTMLElement).scrollIntoView(false);
       } else if (
         !withinviewport(this.element.nativeElement, { sides: 'left top right' })
       ) {
-        (<HTMLElement>this.element.nativeElement).scrollIntoView(true);
+        (this.element.nativeElement as HTMLElement).scrollIntoView(true);
       }
     }
-    (<any>this.opener.trigger)._element = this.element;
+    (this.opener.trigger as any)._element = this.element;
     this.opener.trigger.menu = this.tourStepTemplate.templateComponent.tourStep;
     this.opener.trigger.ngAfterContentInit();
     this.opener.trigger.openMenu();
 
+    step.enableBackdrop = (step.toClick && !step.enableBackdrop) ? true : step.enableBackdrop;
+
     if (step.enableBackdrop) {
       this.tourBackdrop.show(this.element);
     } else {
       this.tourBackdrop.close();
     }
 
+    if (step.toClick) {
+      console.log('Click anywhere to go to the next step.');
+      // TODO Snackbar / Toast with 'Click anywhere to go to the next step' so the User knows
+    }
+
     step.prevBtnTitle = step.prevBtnTitle || 'Prev';
     step.nextBtnTitle = step.nextBtnTitle || 'Next';
     step.endBtnTitle = step.endBtnTitle || 'End';
 
     if (this.menuCloseSubscription) {
       this.menuCloseSubscription.unsubscribe();
     }
+
+    // When the next step is called, if a toClick was defined force click it.
+    // Works for buttons, links, etc if you provide the right element, otherwise just go to next step
+    this.tourService.stepShow$
+      .pipe(first())
+      .subscribe(() => {
```

Comment:
Is this the best place to do it? 
Shouldn't you unsubscribe as well? 
Maybe create a new function, instead of putting all this spaghetti here?

Reply:
This would result in a memory leak as @hakimio , you do not close the subscribtion